pkg/segment/search/searchaggs.go

```diff
@@ -177,10 +177,41 @@ func addRecordToAggregations(grpReq *structs.GroupByRequest, timeHistogram *stru
 	hasLimitOption := false
 	groupByColValCnt := make(map[string]int, 0)
 	var timeRangeBuckets []uint64
+
+	byFieldCnameKeyIdx := int(-1)
+	var isTsCol bool
+	groupbyColKeyIndices := make(map[int]struct{})
+	var byField string
 	if usedByTimechart {
+		byField = timeHistogram.Timechart.ByField
 		timeRangeBuckets = aggregations.GenerateTimeRangeBuckets(timeHistogram)
 		hasLimitOption = timeHistogram.Timechart.LimitExpr != nil
+		cKeyidx, ok := multiColReader.GetColKeyIndex(byField)
+		if ok {
+			byFieldCnameKeyIdx = cKeyidx
+		}
+		if timeHistogram.Timechart.ByField == config.GetTimeStampKey() {
+			isTsCol = true
+		} else {
+			isTsCol = false
+		}
+	} else {
+		for _, col := range grpReq.GroupByColumns {
+			cKeyidx, ok := multiColReader.GetColKeyIndex(col)
+			if ok {
+				groupbyColKeyIndices[cKeyidx] = struct{}{}
+			}
+		}
+	}
+
+	measColKeyIdxAndIndices := make(map[int][]int)
```

Comment:
nit: It might be better to rename this as `measureColKeyIdxAndIndices`

Reply:
done